pkg/controller/controllercmd/cmd.go

```diff
@@ -99,10 +91,25 @@ func (c *ControllerCommandConfig) StartController(ctx context.Context) error {
 		// force the config to our version to read it
 		configCopy.SetGroupVersionKind(operatorv1alpha1.GroupVersion.WithKind("GenericOperatorConfig"))
 		if err := runtime.DefaultUnstructuredConverter.FromUnstructured(configCopy.Object, config); err != nil {
-			return err
+			return nil, nil, nil, err
 		}
 	}
+	return unstructuredConfig, config, configContent, nil
+}
 
+func hasServiceServingCerts(certDir string) bool {
+	if _, err := os.Stat(filepath.Join(certDir, "tls.crt")); os.IsNotExist(err) {
+		return false
+	}
+	if _, err := os.Stat(filepath.Join(certDir, "tls.key")); os.IsNotExist(err) {
+		return false
+	}
+	return true
+}
+
+// StartWithRotation starts the provided builder with the default rotation set (config + serving info). Use StartController if
+// you do not need to customize the controller builder.
+func (c *ControllerCommandConfig) SetDefaultRotation(config *operatorv1alpha1.GenericOperatorConfig, configContent []byte) (map[string][]byte, []string, error) {
```

Comment:
indicate whether this mutates method receiver.  reading top to bottom it is unclear.  The method is called `Set`, but returns values.